server/src/main/java/com/paypal/selion/grid/JarSpawner.java

```diff
@@ -116,317 +84,83 @@ private void printUsageInfo() {
-            cmdLine.addArgument(NODE_CONFIG_ARG).addArgument(NODE_CONFIG_FILE);
-            copyFileFromResources(NODE_CONFIG_FILE_RESOURCE, NODE_CONFIG_FILE);
+        String result = null;
+        if (type.equals(InstanceType.SELENIUM_NODE)) {
+            result = getHost() + ":" + getPort() + "/extra/" + servlet;
         }
-
-        LOGGER.exiting(cmdLine.toString());
-        return cmdLine;
-    }
-
-    /**
-     * Configure SeLion Grid related arguments to the CommandLine
-     *
-     * @param commands
-     *            command line arguments passed from User
-     * @param cmdLine
-     *            the {@link CommandLine} we are building
-     * @return the changed {@link CommandLine}
-     * @throws IOException
-     */
-    private CommandLine addGridArgumentsToCommandLine(List<String> commands, CommandLine cmdLine)
-            throws IOException {
-        LOGGER.entering(new Object[] { commands.toString(), cmdLine.toString() });
-
-        LOGGER.info("This instance is considered a SeLion Grid Hub");
-
-        if (!commands.contains(HUB_CONFIG_ARG)) {
-            String hubConfig = HUB_CONFIG_FILE;
-
-            // To verify this is SeLion Sauce Grid or not
-            if (commands.contains(TYPE_ARG) && commands.contains(SAUCE)) {
-                hubConfig = HUB_SAUCE_CONFIG_FILE;
-                copyFileFromResources(HUB_SAUCE_CONFIG_FILE_RESOURCE, HUB_SAUCE_CONFIG_FILE);
-                copyFileFromResources(SAUCE_CONFIG_FILE_RESOURCE, SAUCE_CONFIG_FILE);
-            } else {
-                copyFileFromResources(HUB_CONFIG_FILE_RESOURCE, HUB_CONFIG_FILE);
-            }
-
-            cmdLine.addArgument(HUB_CONFIG_ARG).addArgument(hubConfig);
+        if (type.equals(InstanceType.SELENIUM_HUB)) {
+            result = getHost() + ":" + getPort() + "/grid/" + servlet;
         }
-
-        LOGGER.exiting(cmdLine.toString());
-        return cmdLine;
+        // standalone
+        LOGGER.exiting(result);
+        return result;
     }
 
-    /**
-     * Create mandatory folders and files required to start the Grid / Node
-     * 
-     * @throws IOException
-     */
-    private void initialSetup() {
-        LOGGER.entering();
-
+    @Override
+    public final void shutdown() {
+        // TODO NodeForceRestartServlet isn't a good option for Hubs
+        // TODO Make process handler consider PID files
+        CloseableHttpClient client = HttpClientBuilder.create().build();
+        StringBuilder url = new StringBuilder();
         try {
-            FileUtils.forceMkdir(new File(DOWNLOADS_DIR));
-            FileUtils.forceMkdir(new File(LOGS_DIR));
-            copyFileFromResources(DOWNLOAD_JSON_FILE_RESOURCE, DOWNLOAD_JSON_FILE);
-            copyFileFromResources(SELION_CONFIG_FILE_RESOURCE, SELION_CONFIG_FILE);
-            ConfigParser.setConfigFile(SELION_CONFIG_FILE);
+            url.append("http://");
+            url.append(getHostPortAndServletPath("NodeForceRestartServlet"));
```

Comment:
Usage of `NodeForceRestartServlet.class.getSimpleName()` requires `HttpServlet` (sub-class of NodeForceRestartServlet) to be added as a `compile` dependency and included into the Selion-Grid (standalone) jar -- which I'm avoiding here. That said, `shutdown()` in all process based `RunnableLaunchers` is the area I called out as needs improvement.. Relying on `NodeForceRestartServlet` here as I have coded, is not a reliable method and one that does not work for `STANDALONE` and `HUB` ...